### PR TITLE
fixed event-visibility in calendar and export function

### PR DESCRIPTION
**Pull request by ysbrandB**
Made the code more readable and fixed events with a publication time.

### CODE REVIEW

File: app/Http/Controllers/SearchController.php [hunk 1]
```diff
@@ -74,7 +74,7 @@ public function search(Request $request)
             ['id', 'title']
         );
         foreach ($presearch_events as $event) {
-            if ($event->mayViewEvent(Auth::user())) {
+            if ($event->mayViewEvent(Auth::user()) || Auth::user()->can('board')) {
```
Comment (Jessseee) [inline app/Http/Controllers/SearchController.php:77]: This shouldn't be necessary, right?

File: app/Http/Controllers/ExportController.php [hunk 2]
```diff
@@ -45,11 +45,11 @@ public function export($table, $personal_key)
...
                 break;
             case 'activities':
-                if ($user->can('admin')) {
+                if ($user->can('board')) {
                     $data = Activity::all();
                 } else {
-                    $data = Activity::with('event')->get()->filter(function ($val) {
-                        return $val->event && $val->event->secret == 0;
+                    $data = Activity::with('event')->get()->filter(function ($activity) use ($user) {
+                        return $activity->event && $activity->event->mayViewEvent($user);
```
Comment (Jessseee) [inline app/Http/Controllers/ExportController.php:52]: I am pretty sure you cannot make an activity without an event, right?

File: app/Models/Event.php [hunk 3]
```diff
@@ -117,13 +117,9 @@ public function committee()
...
     {
-        //board may always view events
-        if($user && $user->can('board')){
-            return true;
-        }
-        //only show secret events if the user is participating
+        //only show secret events if the user is participating, helping or organising
         if($this->secret){
-            if($user && $this->activity && $this->activity->isParticipating($user)){
+            if($user && $this->activity && ($this->activity->isParticipating($user) || $this->activity->isHelping($user) || $this->activity->isOrganising($user))){
```
Comment (Jessseee) [inline app/Models/Event.php:122]: You could also add `$user->can('board')` to this check.

File: resources/views/event/display_includes/event_block.blade.php [hunk 4]
```diff
@@ -1,4 +1,4 @@
-@if($event->mayViewEvent(Auth::user()))
+@if($event->mayViewEvent(Auth::user()) || Auth::user()->can('board'))
```
Comment (Jessseee) [inline resources/views/event/display_includes/event_block.blade.php:1]: Same here, you can add this to mayViewEvent

File: app/Http/Controllers/ExportController.php [hunk 5]
```diff
@@ -70,10 +70,12 @@ public function export($table, $personal_key)
                 $data = Company::all();
                 break;
             case 'events':
-                if ($user->can('admin')) {
+                if ($user->can('sysadmin')) {
```
Comment (Jessseee) [inline app/Http/Controllers/ExportController.php:73]: This should be 'admin' as a user with the 'board' role may also view all events.

**Review by Jessseee — CHANGES_REQUESTED** [review]

File: resources/views/event/display_includes/event_block.blade.php (hunk 4 above)
Comment (ysbrandB) [reply resources/views/event/display_includes/event_block.blade.php:1]: Nope, mayViewEvent is now also used for the calendar and you do want to give the board the possibility to not have all events in their calendar.

**Review by Jessseee — APPROVED** [review]
LGTM

**Pull request merged**